src/node_tracing.cc

```diff
@@ -0,0 +1,102 @@
+#include "node_tracing.h"
+#include "base_object-inl.h"
+#include "env-inl.h"
+#include "memory_tracker-inl.h"
+#include "node.h"
+#include "node_external_reference.h"
+#include "node_internals.h"
+#include "node_v8_platform-inl.h"
+#include "tracing/agent.h"
+#include "util-inl.h"
+#include "v8.h"
+
+#include <numeric>
+#include <set>
+#include <string>
+
+namespace node {
+namespace tracing {
+
+using v8::Context;
+using v8::FunctionTemplate;
+using v8::HandleScope;
+using v8::Local;
+using v8::Object;
+using v8::Uint8Array;
+using v8::Value;
+
+void BindingData::MemoryInfo(MemoryTracker* tracker) const {}
+
+BindingData::BindingData(Realm* realm, v8::Local<v8::Object> object)
+    : SnapshotableObject(realm, object, type_int) {
+  // Get the pointer of the memory for the flag that
+  // store if trace is enabled for http the pointer will
+  // always be the same and if the category does not exist
+  // it creates:
+  // https://github.com/nodejs/node/blob/6bbf2a57fcf33266c5859497f8cc32e1389a358a/deps/v8/src/libplatform/tracing/tracing-controller.cc#L322-L342
+  auto p = const_cast<uint8_t*>(
+      TRACE_EVENT_API_GET_CATEGORY_GROUP_ENABLED("node.http"));
+
+  // NO_OP deleter
+  // the idea of this chunk of code is to construct
+  auto nop = [](void*, size_t, void*) {};
+  auto bs = v8::ArrayBuffer::NewBackingStore(
+      p, 1, nop, nullptr);
+  auto ab =
+      v8::ArrayBuffer::New(realm->isolate(), std::move(bs));
+  v8::Local<Uint8Array> u8 = v8::Uint8Array::New(ab, 0, 1);
+
+  object
+      ->Set(realm->context(),
+            FIXED_ONE_BYTE_STRING(realm->isolate(), "tracingCategories"),
+            u8)
+      .Check();
+}
+
+bool BindingData::PrepareForSerialization(v8::Local<v8::Context> context,
+                                          v8::SnapshotCreator* creator) {
+  return true;
+}
+
+InternalFieldInfoBase* BindingData::Serialize(int index) {
+  DCHECK_EQ(index, BaseObject::kEmbedderType);
+  InternalFieldInfo* info =
+      InternalFieldInfoBase::New<InternalFieldInfo>(type());
+  return info;
+}
+
+void BindingData::Deserialize(v8::Local<v8::Context> context,
+                              v8::Local<v8::Object> holder,
+                              int index,
+                              InternalFieldInfoBase* info) {
+  DCHECK_EQ(index, BaseObject::kEmbedderType);
+  v8::HandleScope scope(context->GetIsolate());
+  Realm* realm = Realm::GetCurrent(context);
+  BindingData* binding = realm->AddBindingData<BindingData>(context, holder);
+  CHECK_NOT_NULL(binding);
+}
+
+void BindingData::CreatePerIsolateProperties(IsolateData* isolate_data,
+                                             Local<FunctionTemplate> ctor) {}
```

Comment:
```suggestion
                                             Local<ObjectTemplate> ctor) {}
```
I think this should fix the compilation error (you would also need to update the `using v8::FunctionTemplate;` at the top)

Reply:
This didn't compile:

```
ninja: Entering directory `out/Release'
[3/215] ACTION icudata: icudata_cc7672230cb6f4a4cb1b079ff599d36f
generating assembly code for /home/h4ad/Projects/opensource/node/out/Release/gen/icudt73.dat
[61/215] CXX obj/test/cctest/cctest.test_node_crypto_env.o
In file included from ../../deps/openssl/openssl/include/openssl/../../../config/./bio_asm.h:11,
                 from ../../deps/openssl/openssl/include/openssl/../../../config/bio.h:4,
                 from ../../deps/openssl/openssl/include/openssl/bio.h:1,
                 from ../../deps/openssl/openssl/include/openssl/../../../config/././archs/linux-x86_64/asm/include/openssl/asn1.h:27,
                 from ../../deps/openssl/openssl/include/openssl/../../../config/./asn1_asm.h:11,
                 from ../../deps/openssl/openssl/include/openssl/../../../config/asn1.h:4,
                 from ../../deps/openssl/openssl/include/openssl/asn1.h:1,
                 from ../../deps/openssl/openssl/include/openssl/dsa.h:46,
                 from ../../src/crypto/crypto_util.h:15,
                 from ../../src/crypto/crypto_keys.h:6,
                 from ../../src/crypto/crypto_cipher.h:6,
                 from ../../src/crypto/crypto_aes.h:6,
                 from ../../src/node_crypto.h:31,
                 from ../../src/crypto/crypto_bio.h:27,
                 from ../../test/cctest/test_node_crypto_env.cc:1:
../../test/cctest/test_node_crypto_env.cc: In member function ‘virtual void NodeCryptoEnv_LoadBIO_Test::TestBody()’:
../../deps/openssl/openssl/include/openssl/../../../config/././archs/linux-x86_64/asm/include/openssl/bio.h:495:26: warning: value computed is not used [-Wunused-value]
  495 | # define BIO_seek(b,ofs) (int)BIO_ctrl(b,BIO_C_FILE_SEEK,ofs,NULL)
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../test/cctest/test_node_crypto_env.cc:26:3: note: in expansion of macro ‘BIO_seek’
   26 |   BIO_seek(bio.get(), 2);
      |   ^~~~~~~~
[208/215] LINK cctest
FAILED: cctest 
c++ -pthread -rdynamic -m64 -Wl,--whole-archive obj/deps/zlib/libzlib.a -Wl,--no-whole-archive -Wl,--whole-archive obj/deps/uv/libuv.a -Wl,--no-whole-archive -Wl,-z,noexecstack -Wl,--whole-archive obj/tools/v8_gypfiles/libv8_snapshot.a -Wl,--no-whole-archive -Wl,-z,relro -Wl,-z,now -Wl,--whole-archive,obj/deps/openssl/libopenssl.a -Wl,--no-whole-archive -pthread -o cctest -Wl,--start-group obj/src/cctest.node_snapshot_stub.o obj/test/cctest/cctest.node_test_fixture.o obj/test/cctest/cctest.test_aliased_buffer.o obj/test/cctest/cctest.test_base64.o obj/test/cctest/cctest.test_base_object_ptr.o obj/test/cctest/cctest.test_node_postmortem_metadata.o obj/test/cctest/cctest.test_environment.o obj/test/cctest/cctest.test_linked_binding.o obj/test/cctest/cctest.test_node_api.o obj/test/cctest/cctest.test_per_process.o obj/test/cctest/cctest.test_platform.o obj/test/cctest/cctest.test_report.o obj/test/cctest/cctest.test_json_utils.o obj/test/cctest/cctest.test_sockaddr.o obj/test/cctest/cctest.test_traced_value.o obj/test/cctest/cctest.test_util.o obj/test/cctest/cctest.test_dataqueue.o obj/test/cctest/cctest.test_crypto_clienthello.o obj/test/cctest/cctest.test_node_crypto.o obj/test/cctest/cctest.test_node_crypto_env.o obj/test/cctest/cctest.test_quic_cid.o obj/test/cctest/cctest.test_quic_tokens.o obj/test/cctest/cctest.test_inspector_socket.o obj/test/cctest/cctest.test_inspector_socket_server.o obj/libnode.a obj/deps/base64/libbase64.a obj/deps/googletest/libgtest.a obj/deps/googletest/libgtest_main.a obj/deps/histogram/libhistogram.a obj/deps/uvwasi/libuvwasi.a obj/deps/simdutf/libsimdutf.a obj/deps/ada/libada.a obj/tools/v8_gypfiles/libv8_snapshot.a obj/tools/v8_gypfiles/libv8_libplatform.a obj/tools/icu/libicui18n.a obj/deps/zlib/libzlib.a obj/deps/llhttp/libllhttp.a obj/deps/cares/libcares.a obj/deps/uv/libuv.a obj/deps/nghttp2/libnghttp2.a obj/deps/brotli/libbrotli.a obj/deps/openssl/libopenssl.a obj/deps/ngtcp2/libngtcp2.a obj/deps/ngtcp2/libnghttp3.a obj/deps/base64/libbase64_ssse3.a obj/deps/base64/libbase64_sse41.a obj/deps/base64/libbase64_sse42.a obj/deps/base64/libbase64_avx.a obj/deps/base64/libbase64_avx2.a obj/tools/icu/libicuucx.a obj/tools/icu/libicudata.a obj/tools/v8_gypfiles/libv8_base_without_compiler.a obj/tools/v8_gypfiles/libv8_libbase.a obj/tools/v8_gypfiles/libv8_zlib.a obj/tools/v8_gypfiles/libv8_compiler.a obj/tools/v8_gypfiles/libv8_turboshaft.a obj/tools/v8_gypfiles/libv8_initializers.a obj/deps/zlib/libzlib_inflate_chunk_simd.a obj/deps/zlib/libzlib_adler32_simd.a obj/deps/zlib/libzlib_crc32_simd.a  -lm -ldl -Wl,--end-group
/usr/bin/ld: obj/src/libnode.node_binding.o: in function `node::binding::RegisterBuiltinBindings()':
node_binding.cc:(.text+0x8fb): undefined reference to `_register_tracing()'
/usr/bin/ld: obj/src/libnode.node_snapshotable.o: in function `node::DeserializeNodeInternalFields(v8::Local<v8::Object>, int, v8::StartupData, void*)':
node_snapshotable.cc:(.text+0x8a08): undefined reference to `node::tracing::BindingData::Deserialize(v8::Local<v8::Context>, v8::Local<v8::Object>, int, node::InternalFieldInfoBase*)'
/usr/bin/ld: obj/src/libnode.node_external_reference.o: in function `node::ExternalReferenceRegistry::ExternalReferenceRegistry()':
node_external_reference.cc:(.text+0x179): undefined reference to `_register_external_reference_tracing(node::ExternalReferenceRegistry*)'
collect2: error: ld returned 1 exit status
[209/215] LINK node_mksnapshot
FAILED: node_mksnapshot 
c++ -pthread -rdynamic -m64 -Wl,--whole-archive obj/deps/zlib/libzlib.a -Wl,--no-whole-archive -Wl,--whole-archive obj/deps/uv/libuv.a -Wl,--no-whole-archive -Wl,-z,noexecstack -Wl,--whole-archive obj/tools/v8_gypfiles/libv8_snapshot.a -Wl,--no-whole-archive -Wl,-z,relro -Wl,-z,now -Wl,--whole-archive,obj/deps/openssl/libopenssl.a -Wl,--no-whole-archive -pthread -o node_mksnapshot -Wl,--start-group obj/src/node_mksnapshot.node_snapshot_stub.o obj/tools/snapshot/node_mksnapshot.node_mksnapshot.o obj/libnode.a obj/deps/histogram/libhistogram.a obj/deps/uvwasi/libuvwasi.a obj/deps/ada/libada.a obj/tools/v8_gypfiles/libv8_snapshot.a obj/tools/v8_gypfiles/libv8_libplatform.a obj/tools/icu/libicui18n.a obj/deps/zlib/libzlib.a obj/deps/llhttp/libllhttp.a obj/deps/cares/libcares.a obj/deps/uv/libuv.a obj/deps/nghttp2/libnghttp2.a obj/deps/brotli/libbrotli.a obj/deps/openssl/libopenssl.a obj/deps/ngtcp2/libngtcp2.a obj/deps/ngtcp2/libnghttp3.a obj/deps/base64/libbase64.a obj/deps/base64/libbase64_ssse3.a obj/deps/base64/libbase64_sse41.a obj/deps/base64/libbase64_sse42.a obj/deps/base64/libbase64_avx.a obj/deps/base64/libbase64_avx2.a obj/deps/simdutf/libsimdutf.a obj/tools/icu/libicuucx.a obj/tools/icu/libicudata.a obj/tools/v8_gypfiles/libv8_base_without_compiler.a obj/tools/v8_gypfiles/libv8_libbase.a obj/tools/v8_gypfiles/libv8_zlib.a obj/tools/v8_gypfiles/libv8_compiler.a obj/tools/v8_gypfiles/libv8_turboshaft.a obj/tools/v8_gypfiles/libv8_initializers.a obj/deps/zlib/libzlib_inflate_chunk_simd.a obj/deps/zlib/libzlib_adler32_simd.a obj/deps/zlib/libzlib_crc32_simd.a  -lm -ldl -Wl,--end-group
/usr/bin/ld: obj/src/libnode.node_binding.o: in function `node::binding::RegisterBuiltinBindings()':
node_binding.cc:(.text+0x8fb): undefined reference to `_register_tracing()'
/usr/bin/ld: obj/src/libnode.node_snapshotable.o: in function `node::DeserializeNodeInternalFields(v8::Local<v8::Object>, int, v8::StartupData, void*)':
node_snapshotable.cc:(.text+0x8a08): undefined reference to `node::tracing::BindingData::Deserialize(v8::Local<v8::Context>, v8::Local<v8::Object>, int, node::InternalFieldInfoBase*)'
/usr/bin/ld: obj/src/libnode.node_external_reference.o: in function `node::ExternalReferenceRegistry::ExternalReferenceRegistry()':
node_external_reference.cc:(.text+0x179): undefined reference to `_register_external_reference_tracing(node::ExternalReferenceRegistry*)'
collect2: error: ld returned 1 exit status
[210/215] LINK embedtest
FAILED: embedtest 
c++ -pthread -rdynamic -m64 -Wl,--whole-archive obj/deps/zlib/libzlib.a -Wl,--no-whole-archive -Wl,--whole-archive obj/deps/uv/libuv.a -Wl,--no-whole-archive -Wl,-z,noexecstack -Wl,--whole-archive obj/tools/v8_gypfiles/libv8_snapshot.a -Wl,--no-whole-archive -Wl,-z,relro -Wl,-z,now -Wl,--whole-archive,obj/deps/openssl/libopenssl.a -Wl,--no-whole-archive -pthread -o embedtest -Wl,--start-group obj/src/embedtest.node_snapshot_stub.o obj/test/embedding/embedtest.embedtest.o obj/libnode.a obj/deps/histogram/libhistogram.a obj/deps/uvwasi/libuvwasi.a obj/deps/ada/libada.a obj/tools/v8_gypfiles/libv8_snapshot.a obj/tools/v8_gypfiles/libv8_libplatform.a obj/tools/icu/libicui18n.a obj/deps/zlib/libzlib.a obj/deps/llhttp/libllhttp.a obj/deps/cares/libcares.a obj/deps/uv/libuv.a obj/deps/nghttp2/libnghttp2.a obj/deps/brotli/libbrotli.a obj/deps/openssl/libopenssl.a obj/deps/ngtcp2/libngtcp2.a obj/deps/ngtcp2/libnghttp3.a obj/deps/base64/libbase64.a obj/deps/base64/libbase64_ssse3.a obj/deps/base64/libbase64_sse41.a obj/deps/base64/libbase64_sse42.a obj/deps/base64/libbase64_avx.a obj/deps/base64/libbase64_avx2.a obj/deps/simdutf/libsimdutf.a obj/tools/icu/libicuucx.a obj/tools/icu/libicudata.a obj/tools/v8_gypfiles/libv8_base_without_compiler.a obj/tools/v8_gypfiles/libv8_libbase.a obj/tools/v8_gypfiles/libv8_zlib.a obj/tools/v8_gypfiles/libv8_compiler.a obj/tools/v8_gypfiles/libv8_turboshaft.a obj/tools/v8_gypfiles/libv8_initializers.a obj/deps/zlib/libzlib_inflate_chunk_simd.a obj/deps/zlib/libzlib_adler32_simd.a obj/deps/zlib/libzlib_crc32_simd.a  -lm -ldl -Wl,--end-group
/usr/bin/ld: obj/src/libnode.node_binding.o: in function `node::binding::RegisterBuiltinBindings()':
node_binding.cc:(.text+0x8fb): undefined reference to `_register_tracing()'
/usr/bin/ld: obj/src/libnode.node_snapshotable.o: in function `node::DeserializeNodeInternalFields(v8::Local<v8::Object>, int, v8::StartupData, void*)':
node_snapshotable.cc:(.text+0x8a08): undefined reference to `node::tracing::BindingData::Deserialize(v8::Local<v8::Context>, v8::Local<v8::Object>, int, node::InternalFieldInfoBase*)'
/usr/bin/ld: obj/src/libnode.node_external_reference.o: in function `node::ExternalReferenceRegistry::ExternalReferenceRegistry()':
node_external_reference.cc:(.text+0x179): undefined reference to `_register_external_reference_tracing(node::ExternalReferenceRegistry*)'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```